SpatialGDK/Source/SpatialGDKServices/Private/LocalDeploymentManager.cpp

```diff
@@ -351,6 +358,65 @@ bool FLocalDeploymentManager::SetupRuntimeFileLogger(const FString& RuntimeLogDi
 }
 
 bool FLocalDeploymentManager::TryStopLocalDeployment()
+{
+	if (!StartLocalDeploymentShutDown())
+	{
+		return false;
+	}
+
+	RuntimeProcess->Stop();
+
+	bool bRuntimeShutDownSuccesfully = WaitForRuntimeProcessToShutDown();
+	FinishLocalDeploymentShutDown();
+
+	return bRuntimeShutDownSuccesfully;
+}
+
+bool FLocalDeploymentManager::TryStopLocalDeploymentGracefully()
+{
+	if (bTestRunnning)
+	{
+		bTestRunnning = false;
+		return TryStopLocalDeployment();
+	}
+
+	if (!StartLocalDeploymentShutDown())
+	{
+		return false;
+	}
+
+	FHttpModule& HttpModule = FModuleManager::LoadModuleChecked<FHttpModule>("HTTP");
+#if ENGINE_MINOR_VERSION >= 26
+	TSharedRef<IHttpRequest, ESPMode::ThreadSafe> HttpRequest = HttpModule.Get().CreateRequest();
+#else
+	TSharedRef<IHttpRequest> HttpRequest = HttpModule.Get().CreateRequest();
+#endif
+
+	HttpRequest->SetURL(TEXT("http://localhost:5006/shutdown"));
```

Comment:
we have similar code for taking a snapshot
may be worth refactoring out into a runtimeHttpEndpoint function at some point